ako-gateway-api/nodes/avi_model_l7_translator.go

```diff
@@ -244,7 +245,27 @@ func (o *AviObjectGraph) BuildPGPool(key, parentNsName string, childVsNode *node
 		Name:   PGName,
 		Tenant: lib.GetTenant(),
 	}
+	var resolvedRefCondition akogatewayapistatus.Condition
+	hasInvalidBackend := false
+	_, namespace, name := lib.ExtractTypeNameNamespace(routeTypeNsName)
+	httpRoute, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().HTTPRouteInformer.Lister().HTTPRoutes(namespace).Get(name)
+	if err != nil {
+		utils.AviLog.Warnf("key: %s, msg: Unable to extract the HTTPRoute object %s for BackendRef validation", key, name)
+		return
+	}
+
 	for _, httpbackend := range rule.Backends {
+		isValidBackend, resolvedRefConditionforBackend := validateBackendReference(key, *httpbackend.Backend, httpRoute)
```

Comment:
I think we should include this logic as part of `ParseRouteConfig` instead of writing down over here. Advantage: We will separate of parsing (and setting of httproute object) from logic of populating AviController objects. Also we will save one Informer call.

Reply:
Done